src/status_im/navigation/effects.cljs

```diff
@@ -122,9 +122,11 @@
 (defn dismiss-modal
   ([] (dismiss-modal nil))
   ([comp-id]
-   (reset! state/dissmissing true)
-   (navigation/dismiss-modal (name (or comp-id (last @state/modals))))
-   (state/navigation-pop-from comp-id)))
+   (when (or (not comp-id)
+             (state/modal-open? comp-id))
+     (reset! state/dissmissing true)
+     (navigation/dismiss-modal (name (or comp-id (last @state/modals))))
+     (state/navigation-pop-from comp-id))))
```

Comment:
Why do we need to change root level navigation?

Reply:
the only change there is a condition to check whether the the modal id we pass to `:close-modal` is actually present, because if it's not, it breaks the app and the user can't navigate anywhere. It happens sometimes in our case, but unsure why or when.

Reply:
the only change there is a condition to check whether the the modal id we pass to :close-modal is actually present, because if it's not, it breaks the app and the user can't navigate anywhere. Since we have the unmount effect, it will fire when the user presses the X button as well. Not ideal though, I agree.